delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/OrganizationsDummy.scala

```diff
@@ -58,28 +57,17 @@ final class OrganizationsDummy private (
   )(implicit caller: Subject): IO[OrganizationRejection, OrganizationResource] =
     eval(DeprecateOrganization(label, rev, caller))
 
-  override def fetch(label: Label): UIO[Option[OrganizationResource]] =
-    cache.fetch(label)
+  override def fetch(label: Label): IO[OrganizationNotFound, OrganizationResource] =
+    cache.fetch(label).flatMap(IO.fromOption(_, OrganizationNotFound(label)))
 
-  override def fetchAt(
-      label: Label,
-      rev: Long
-  ): IO[OrganizationRejection.RevisionNotFound, Option[OrganizationResource]] =
+  override def fetchAt(label: Label, rev: Long): IO[OrganizationRejection, OrganizationResource] =
```

Comment:
I guess it is ok but one drawback of this PR is that some fetch/read operations return a large possible range of rejections when 2 or 3 of them actually matter

Reply:
That's true. However is the case already for the rest of signatures.